### DISCUSSION TITLE
Domain Decomposition

### DISCUSSION THREAD

**Discussion by HamidTGanji** [lululxvi/deepxde#1322] [Q&A]
Dear LuLu.
Thank you for providing such a great package.
I'm trying to solve the 1D-Wave equation in a bar with two different materials. Let's assume that c=1 for 0<x<0.7 and c=2 for 0.7<x<1.
For this problem, we need continuous displacement and force at the interface of two materials. However, when I solved this problem with only 1 domain, I noticed that the second condition (i.e. continuity of stress) was not satisfied. Based on my searches, it seems that I should use domain decomposition in order to solve these problems with discontinuity. So, I should have 2 networks and introduce the boundary condition of these networks to be a function of the other network. How can I do that? In the attached picture, u_1 is the solution of 1st subdomain which comes from the 1st network and u_2 is the solution of second subdomain which comes from the 2nd network.

![DomainDecomposition](https://github.com/lululxvi/deepxde/assets/126012818/c3e01a42-ec1b-4f0c-8213-25f6c700ee0b)

**Comment by lululxvi** [comment]
Yes, you should have 2 networks and introduce the boundary condition of these networks. What is your specific question?

**Reply by HamidTGanji** [reply]
Thank you for your reply professor.
My problem is that the boundary conditions are not predefined known functions. Actually, the BCs of each network are functions of the output of the other network. How can we define these types of networks?

**Reply by lululxvi** [reply]
Use `OperatorBC`
